crates/matrix-sdk/src/client/mod.rs

```diff
@@ -798,8 +806,11 @@ impl Client {
     /// [`room::Room`], [`Client`] for now.
     pub async fn register_notification_handler<H, Fut>(&self, handler: H) -> &Self
     where
-        H: Fn(Notification, room::Room, Client) -> Fut + Send + Sync + 'static,
-        Fut: Future<Output = ()> + Send + 'static,
+        H: Fn(Notification, room::Room, Client) -> Fut
+            + SendOutsideWasm
```

Comment:
@dkasak had a nice idea for that trait, `SendUnlessWasm`, sounds a bit better than the `Outside`.

Reply:
That was the [initial name](https://github.com/matrix-org/matrix-rust-sdk/pull/932#issuecomment-1208332386) :D, not sure which one is better.